homeassistant/components/weather/metoffice.py

```diff
@@ -0,0 +1,117 @@
+"""
+Support for UK Met Office weather service.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.metoffice/
+"""
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.weather import WeatherEntity, PLATFORM_SCHEMA
+from homeassistant.const import CONF_NAME, TEMP_CELSIUS
+from homeassistant.helpers import config_validation as cv
+# Reuse data and API logic from the sensor implementation
```

Comment:
From my point of view it should be the other way around. The `weather` platform should act as the primary implementation and the `sensor` platform as a dependent.